extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/dashmanifestcreators/YoutubeDashManifestCreatorsUtils.java

```diff
@@ -0,0 +1,856 @@
+ *
+ * <p>
+ * This class includes common methods of manifest creators and useful constants.
+ * </p>
+ *
+ * <p>
+ * Generation of DASH documents and their conversion as a string is done using external classes
+ * from {@link org.w3c.dom} and {@link javax.xml} packages.
+ * </p>
+ */
+public final class YoutubeDashManifestCreatorsUtils {
+
+    private YoutubeDashManifestCreatorsUtils() {
+    }
+
+    /**
+     * The redirect count limit that this class uses, which is the same limit as OkHttp.
+     */
+    public static final int MAXIMUM_REDIRECT_COUNT = 20;
+
+    /**
+     * URL parameter of the first sequence for live, post-live-DVR and OTF streams.
+     */
+    public static final String SQ_0 = "&sq=0";
+
+    /**
+     * URL parameter of the first stream request made by official clients.
+     */
+    public static final String RN_0 = "&rn=0";
+
+    /**
+     * URL parameter specific to web clients. When this param is added, if a redirection occurs,
+     * the server will not redirect clients to the redirect URL. Instead, it will provide this URL
+     * as the response body.
+     */
+    public static final String ALR_YES = "&alr=yes";
+
+    /**
+     * Constant which represents the {@code MPD} element of DASH manifests.
+     */
+    public static final String MPD = "MPD";
+
+    /**
+     * Constant which represents the {@code Period} element of DASH manifests.
+     */
+    public static final String PERIOD = "Period";
+
+    /**
+     * Constant which represents the {@code AdaptationSet} element of DASH manifests.
+     */
+    public static final String ADAPTATION_SET = "AdaptationSet";
+
+    /**
+     * Constant which represents the {@code Role} element of DASH manifests.
+     */
+    public static final String ROLE = "Role";
+
+    /**
+     * Constant which represents the {@code Representation} element of DASH manifests.
+     */
+    public static final String REPRESENTATION = "Representation";
+
+    /**
+     * Constant which represents the {@code AudioChannelConfiguration} element of DASH manifests.
+     */
+    public static final String AUDIO_CHANNEL_CONFIGURATION = "AudioChannelConfiguration";
+
+    /**
+     * Constant which represents the {@code SegmentTemplate} element of DASH manifests.
+     */
+    public static final String SEGMENT_TEMPLATE = "SegmentTemplate";
+
+    /**
+     * Constant which represents the {@code SegmentTimeline} element of DASH manifests.
+     */
+    public static final String SEGMENT_TIMELINE = "SegmentTimeline";
+
+    /**
+     * Constant which represents the {@code SegmentBase} element of DASH manifests.
+     */
+    public static final String BASE_URL = "BaseURL";
+
+    /**
+     * Constant which represents the {@code SegmentBase} element of DASH manifests.
+     */
+    public static final String SEGMENT_BASE = "SegmentBase";
+
+    /**
+     * Constant which represents the {@code Initialization} element of DASH manifests.
+     */
```

Comment:
I don't know if these are just too many comments ;-)
I would rather put all `public static final String ...` lines together with at the top `// These are elements of DASH manifests` to reduce duplication and redundancy.
**Do not do anything about this**, it is just personal opinion. If you want to join the discussion you are welcome @litetex ;-)

Reply:
Having to agree with Stypox here.
Maybe we could add some kind of link to the structure of DASH in general? Isn't it documented somewhere how a DASH manifest has to look? (quick search found [this](https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_delivery/Setting_up_adaptive_streaming_media_sources))

Reply:
There was one in `DashMpdParser`, but the class has been removed ;-)

Reply:
Ok, then I'm applying this here